apps/framework-cli/src/infrastructure/redis/redis_client.rs

```diff
@@ -399,6 +399,20 @@ impl RedisClient {
             .context("Failed to ping Redis server")?;
         Ok(())
     }
+
+    pub async fn set<K: ToRedisArgs + Send + Sync, V: ToRedisArgs + Send + Sync>(
```

Comment:
I'm concerned about this function.  It accepts a key that may not strictly adhere to the naming conventions established by the RedisClient. I.e; this may break namespacing, which may complicate cleanup.

Reply:
Also, I think this function should accept an optional value for key expiration - which will help with cleanup.

Reply:
Yeah we either have the keys for all operations in the `RedisClient` and hide the `set` function, or we expose the `set` function and hope they use the `prefix` functions.

Reply:
Not sure.  I think the prefix in the RedisClient establishes a global namespace partition for all the keys that Moose uses.  You could have a generic function in RedisClient that performs the desired set operation that includes, via function signature, a string that appends one or more application-specific key segments.  See how `register_lock` works.